tests/src/test/scala/org/apache/openwhisk/core/database/cosmosdb/CosmosDBArtifactStoreTests.scala

```diff
@@ -27,6 +29,27 @@ import org.apache.openwhisk.core.database.test.behavior.ArtifactStoreBehavior
 class CosmosDBArtifactStoreTests extends FlatSpec with CosmosDBStoreBehaviorBase with ArtifactStoreBehavior {
   override protected def maxAttachmentSizeWithoutAttachmentStore = 1.MB
 
+  private var initialLevel: Level = _
+
+  override protected def beforeAll(): Unit = {
+    RecordingLeakDetectorFactory.register()
```

Comment:
Is there any reason not to register this leak detector via the CosmosDBArtifactStore? So that if someone enabled the appropriate leak detection level via system properties, the artifact store can be debugged in the openwhisk system, not just in the test?

Reply:
Discussed with Chetan - this is not necessary to enable leak detection when running in openwhisk, just for being able to assert the test did not cause any leaks.